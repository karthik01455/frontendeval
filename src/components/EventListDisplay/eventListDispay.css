.el-container {
  height: 90vh;
  display: flex;
  flex-direction: column;
}
.el-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.el-left {
  display: flex;
  margin-top: 5%;
}
.el-right {
  display: flex;
  flex-direction: column;
  margin-top: 5%;
}
.el-search {
}
.el-list-container {
  display: flex;
  flex-direction: row;
  column-gap: 10px;
  margin-top: 5%;
  justify-content: center;
  flex-wrap: wrap;
  overflow: scroll;
  height: 70vh;
}
.el-bookmarks {
  display: flex;
  flex-direction: column;

  align-items: end;
}
.el-search {
  display: flex;
  flex-direction: column;
}
.el-left {
  display: flex;
  height: 100px;
  width: 500px;
}
