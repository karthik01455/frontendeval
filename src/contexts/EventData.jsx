/* eslint-disable react/prop-types */
import React, { createContext, useState } from 'react';

export const EventDataContext = createContext({});

export function EventDataProvider({ children }) {
  const [eventData, setEventData] = useState(null);
  const [search, setSearch] = useState(null);

  return (
    <EventDataContext.Provider
      value={{
        eventData,
        setEventData,
        search,
        setSearch,
      }}
    >
      {children}
    </EventDataContext.Provider>
  );
}
