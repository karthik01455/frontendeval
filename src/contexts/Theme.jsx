/* eslint-disable react/prop-types */
import React, { createContext, useState } from 'react';

export const ThemeContext = createContext({});

export function ThemeProvider({ children }) {
  const [theme, setTheme] = useState(null);

  return (
    <ThemeContext.Provider
      value={{
        theme,
        setTheme,
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
}
