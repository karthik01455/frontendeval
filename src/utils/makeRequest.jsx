import axios from 'axios';
import { BACKEND_URL } from '../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig = {}, navigate) => {
  const requestDetails = {
    baseURL: BACKEND_URL,
    url: apiEndPoint.url,
    method: apiEndPoint.method,
    ...dynamicConfig,
    // headers: {
    //   authorization: "Bearer QWlzaHdhcnlhIE4=",
    // },
  };
  try {
    console.log(`req${JSON.stringify(requestDetails)}`);
    const { data } = await axios(requestDetails);
    console.log('hi');
    console.log(`data-mr${JSON.stringify(data)}`);
    return data;
  } catch (e) {
    if (navigate) {
      console.log('es1', e.response);
      const errorStatus = e.response?.status;
      console.log('es', errorStatus);
      if (errorStatus) {
        navigate(`/error/${errorStatus}`);
      } else {
        navigate('/error');
      }
    }
  }
};
export default makeRequest;
