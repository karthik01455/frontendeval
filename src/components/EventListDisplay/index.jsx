import React, { useContext, useEffect, useState } from 'react';
import './eventListDispay.css';
import '../../styles/padding.css';
import { EventDataContext } from '../../contexts/EventData';
import makeRequest from '../../utils/makeRequest';
import EventListCards from '../EventListCards';
import DropDown from '../DropDown';
import {
  GET_EVENTS,
  PATCH_EVENTS,
  GET_EVENTS_ID,
  GET_THEME,
  PUT_THEME,
} from '../../constants/apiEndPoints';
export default function EventListDisplay() {
  const { eventData, setEventData } = useContext(EventDataContext);

  const { event, setEvent } = useContext(EventDataContext);
  const handleSearchChange = (event) => {
    setSearch(event.target.value);
  };
  const [filter, setFilter] = useState('All');
  const [filterBookMarked, setFilterBookMarked] = useState(false);
  const [filterSeatsAvailable, setFilterSeatsAvailable] = useState(false);
  const handleFilterChange = (event) => {
    setFilter(event.target.value);
  };
  const [search, setSearch] = useState(null);
  useEffect(() => {
    makeRequest(GET_EVENTS).then((res) => {
      setEventData(res);
      console.log(res);
    });
  }, []);
  const options = [
    {
      value: 'All',
      label: 'All',
    },
    {
      value: 'Registered',
      label: 'Registered',
    },
  ];
  let displayData = eventData;
  displayData =
    eventData && eventData.length
      ? displayData.sort((a, b) => {
          if (a.datetime < b.datetime) {
            return -1;
          }
        })
      : [];
  displayData =
    search && search !== ''
      ? displayData.filter((value) => {
          return value.name.includes(search);
        })
      : displayData;
  displayData =
    filter && filter !== 'All'
      ? displayData.filter((value) => {
          return value.isRegistered === true;
        })
      : displayData;
  displayData = filterBookMarked
    ? displayData.filter((value) => {
        return value.isBookmarked === true;
      })
    : displayData;
  displayData = filterSeatsAvailable
    ? displayData.filter((value) => {
        return value.areSeatsAvailable === true;
      })
    : displayData;

  return displayData ? (
    <div className='el-container padding'>
      <div className='el-header'>
        <div className='el-left'>
          <DropDown options={options} onChange={handleFilterChange}></DropDown>
        </div>
        <div className='el-right'>
          <div className='el-search'>
            <div className='search-bar'>
              <input value={search} onChange={handleSearchChange}></input>
              {'event'}
              {JSON.stringify(event)}
            </div>
            <div className='el-bookmarks'>
              BookMarked{' '}
              <input
                type='radio'
                name='bookmark'
                id='bookmark'
                checked={filterBookMarked}
                onClick={() => {
                  setFilterBookMarked(!filterBookMarked);
                }}
                onChange={() => {
                  setFilterBookMarked(!filterBookMarked);
                }}
              />
              Seats Available{' '}
              <input
                type='radio'
                name='seats'
                id='seats'
                checked={filterSeatsAvailable}
                onClick={() => {
                  setFilterSeatsAvailable(!filterSeatsAvailable);
                }}
                onChange={() => {
                  setFilterSeatsAvailable(!filterSeatsAvailable);
                }}
              />
            </div>
          </div>
        </div>
      </div>
      <div className='el-list-container'>
        {displayData.map((value) => {
          return (
            // eslint-disable-next-line react/jsx-key
            <EventListCards value={value} key={value.id} />
          );
        })}
      </div>
    </div>
  ) : (
    <div>
      <p>Loading...</p>
    </div>
  );
}
