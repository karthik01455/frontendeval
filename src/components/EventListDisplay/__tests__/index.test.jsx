import React from 'react';
import { fireEvent, screen, render, waitFor } from '@testing-library/react';
import EventListDisplay from '..';
import makeRequest from '../../../utils/makeRequest';
import { mockData } from '../../../mocks/mockData';
import { ThemeContext } from '../../../contexts/Theme';
import { EventDataContext } from '../../../contexts/EventData';
const mockedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockedNavigate,
}));
jest.mock('../../../utils/makeRequest');
describe('EvenListDisplay', () => {
  test('should render the same snapshot when it is called ', () => {
    makeRequest.mockResolvedValue(mockData);
    const { container } = render(
      <EventDataContext.Provider
        value={{
          eventData: mockData,
          setEventData: jest.fn(),
        }}
      >
        <EventListDisplay />
      </EventDataContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
describe('EvenListDisplay', () => {
  test('should render the cards when it eventData is set ', () => {
    makeRequest.mockResolvedValue(mockData);
    const { container } = render(
      <EventDataContext.Provider
        value={{
          eventData: mockData,
          setEventData: jest.fn(),
        }}
      >
        <EventListDisplay />
      </EventDataContext.Provider>
    );
    expect(screen.getAllByTestId('card')).toBeTruthy();
  });
});
