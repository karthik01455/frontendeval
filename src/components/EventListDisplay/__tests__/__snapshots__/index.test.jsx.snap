// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvenListDisplay should render the same snapshot when it is called  1`] = `
<div>
  <div
    class="el-container padding"
  >
    <div
      class="el-header"
    >
      <div
        class="el-left"
      >
        <div>
          <select
            class="drop-down-container"
          >
            <option
              class="options"
              value="All"
            >
              <div
                class="options"
              >
                All
              </div>
            </option>
            <option
              class="options"
              value="Registered"
            >
              <div
                class="options"
              >
                Registered
              </div>
            </option>
          </select>
        </div>
      </div>
      <div
        class="el-right"
      >
        <div
          class="el-search"
        >
          <div
            class="search-bar"
          >
            <input
              value=""
            />
          </div>
          <div
            class="el-bookmarks"
          >
            <div>
               
              BookMarked
               
              <input
                id="bookmark"
                name="bookmark"
                type="radio"
              />
            </div>
            <div>
               
              Seats Available
               
              <input
                id="seats"
                name="seats"
                type="radio"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="el-list-container"
    >
      <div
        class="el-card"
        data-testid="card"
      >
        <div
          class="el-image-container"
        >
          <img
            class="el-image"
            src="https://i.ibb.co/3zbdvWX/battle-of-bands.jpg"
          />
        </div>
        <div
          class="el-card-text"
        >
          <div
            class="el-card-name"
          >
            Battle of the Bands
          </div>
          <div
            class="el-card-description"
          >
            Get ready for Battle of the Bands, where the hottest up-and-coming rock groups will compete for the ultimate prize. With heart-pumping beats and electrifying performances, you won't want to miss this adrenaline-fueled event. Each band will bring their A-game, leaving everything on the stage for the chance to be crowned champion. Don't miss out on the most unforgettable rock competition of the year!
          </div>
          <div
            class="el-card-text-footer"
          >
            <div>
              VENUE:
              All Stars Arena, Las Vegas, NV, USA
            </div>
            <div>
              DATE:
              1
               
              Mar
               
              2023
               
              1677646800000
            </div>
          </div>
        </div>
        <div
          class="el-card-footer"
        >
          <div
            class="el-card-registered"
          >
            <div
              class="el-card-seat"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-check "
                data-icon="circle-check"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div />
          </div>
          <div
            class="el-card-bookmark"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-book-bookmark "
              data-icon="book-bookmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 96C0 43 43 0 96 0h96V190.7c0 13.4 15.5 20.9 26 12.5L272 160l54 43.2c10.5 8.4 26 .9 26-12.5V0h32 32c17.7 0 32 14.3 32 32V352c0 17.7-14.3 32-32 32v64c17.7 0 32 14.3 32 32s-14.3 32-32 32H384 96c-53 0-96-43-96-96V96zM64 416c0 17.7 14.3 32 32 32H352V384H96c-17.7 0-32 14.3-32 32z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="el-card"
        data-testid="card"
      >
        <div
          class="el-image-container"
        >
          <img
            class="el-image"
            src="https://thumbs.dreamstime.com/b/rodeo-cowboy-rough-ride-tossed-around-his-horse-red-bluff-california-april-63692080.jpg"
          />
        </div>
        <div
          class="el-card-text"
        >
          <div
            class="el-card-name"
          >
            Cowboy Rodeo
          </div>
          <div
            class="el-card-description"
          >
            Hold on tight for the Cowboy Rodeo, where the best riders in the country will compete for the championship title. Witness heart-stopping events like bull riding, calf roping, and steer wrestling that showcase the strength and bravery of these incredible cowboys. With non-stop action and adrenaline-fueled excitement, the Cowboy Rodeo is the ultimate display of rodeo skills and cowboy grit. Don't miss out on this wild and exhilarating event!
          </div>
          <div
            class="el-card-text-footer"
          >
            <div>
              VENUE:
              BB Center, Dallas, TX, USA
            </div>
            <div>
              DATE:
              2
               
              Mar
               
              2023
               
              1677726000000
            </div>
          </div>
        </div>
        <div
          class="el-card-footer"
        >
          <div
            class="el-card-registered"
          >
            <div
              class="el-card-seat"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-xmark "
                data-icon="circle-xmark"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div>
              NO SEATS AVAILABLE
            </div>
          </div>
          <div
            class="el-card-bookmark"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-book-bookmark "
              data-icon="book-bookmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 96C0 43 43 0 96 0h96V190.7c0 13.4 15.5 20.9 26 12.5L272 160l54 43.2c10.5 8.4 26 .9 26-12.5V0h32 32c17.7 0 32 14.3 32 32V352c0 17.7-14.3 32-32 32v64c17.7 0 32 14.3 32 32s-14.3 32-32 32H384 96c-53 0-96-43-96-96V96zM64 416c0 17.7 14.3 32 32 32H352V384H96c-17.7 0-32 14.3-32 32z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
